assets/src/legacy/filter.js

```diff
@@ -392,48 +392,57 @@ var lizLayerFilterTool = function () {
                     fieldname: field,
                     filter: ''
                 };
-                $.get(globalThis['filterConfigData'].url, sdata, function (result) {
-                    if (!checkResult(result)) {
-                        return false;
-                    }
-
-                    var autocompleteData = [];
-                    for (var a in result) {
-                        var feat = result[a];
-                        if (feat['v'] === null || !feat['v'] || (typeof feat['v'] === 'string' && feat['v'].trim() === ''))
-                            continue;
-                        autocompleteData.push(feat['v']);
-                    }
-
-                    var html = '';
-                    html += getFormFieldHeader(field_item);
-                    html += '<div style="width: 100%;">'
-                    html += '<input id="liz-filter-field-text' + lizMap.cleanName(field_item.title) + '" class="liz-filter-field-text" value="" title="' + lizDict['filter.input.text.title'] + '" placeholder="' + lizDict['filter.input.text.placeholder'] + '">';
-                    html += '</div>'
-                    html += getFormFieldFooter(field_item);
-
-                    $("#filter-field-order" + String(field_item.order)).append(html);
-                    addFieldEvents(field_item);
 
-                    $("#liz-filter-field-text" + lizMap.cleanName(field_item.title)).autocomplete({
-                        source: autocompleteData,
-                        autoFocus: false, // do not autofocus, because this prevents from searching with LIKE
-                        delay: 200,
-                        minLength: 2,
-                        select: function (event, ui) {
-                            $(this).val(ui.item.value);
-                            $(this).change();
+                var html = '';
+                html += getFormFieldHeader(field_item);
+                html += '<div style="width: 100%;">'
+                html += '<input id="liz-filter-field-text' + lizMap.cleanName(field_item.title) + '" class="liz-filter-field-text" value="" title="' + lizDict['filter.input.text.title'] + '" placeholder="' + lizDict['filter.input.text.placeholder'] + '">';
+                html += '</div>'
+                html += getFormFieldFooter(field_item);
+
+                $("#filter-field-order" + String(field_item.order)).append(html);
+                addFieldEvents(field_item);
+
+                $("#liz-filter-field-text" + lizMap.cleanName(field_item.title)).autocomplete({
+                    source: function (request, response) {
+                        const autocompleteFilter = `"${field}" ILIKE '%${request.term}%' `;
+                        if (!globalThis['filterConfigData'].filter) {
+                            sdata.filter = autocompleteFilter;
+                        } else {
+                            sdata.filter = globalThis['filterConfigData'].filter + ' AND ' + autocompleteFilter;
```

Comment:
The generated filter can contain OR terms. We should use parenthesis to ensure the correct logic will be used. Spaces are used around them to please QGIS :
```suggestion
                            sdata.filter = ' ( ' + globalThis['filterConfigData'].filter + ' ) AND ' + autocompleteFilter;
```